src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemTokenFilter.java

```diff
@@ -19,13 +19,12 @@
  * Greek stemmer.
  */
 public class SkroutzGreekStemTokenFilter extends TokenFilter {
-  private final SkroutzGreekStemmer stemmer;
```

Comment:
Is this change necessary here?

Reply:
No. But I don't see why to keep this initialization in the constructor as far as it does not depend on the `input` parameter (nor on `termAtt` and `keywordAttr`). It seems nice to me to give it an equal place in the initializations chain... Truth is that maybe this change should follow the removal of the stemming exceptions functionality on a separate commit; conceptually it does not really belong here :) Would you prefer to revert this?

Reply:
I'm afraid I don't have enough context to judge this change, but for what I can tell the behavior is not strictly the same. Prior to the change each `SkroutzGreekStemTokenFilter` instance would initialize its own `SkroutzGreekStemmer` instance. Now they will all share one. I don't know whether we care, I just wanted to point it out :)

By the way, is there something wrong with the indentation? 
You can put it in a separate commit if you want, it's up to you.

Reply:
Hmm... No, in Java you need the `static` modifier to have a "class member" (i.e. common for all objects of this class). As the constructor is not a method in Java, you have to use `this` there in any case. By taking the initialization of the variable out of the constructor nothing changes regarding the kind of the variable. The only thing it does change here is the order of the initializations. Inline initializations go first, followed by the code in the constructor (in fact, we can imagine it as if all inline initialization are put in order at the top of the constructor body). So, in our case we have the initializations taking place in this order:
`stemmer`, `termAtt`, `keywordAttr`, <whatever else `super` initializes`,
while previously we had:
`termAtt`, `keywordAttr`, <whatever else `super` initializes`, `stemmer`
In the absence of dependencies between the initializations, all these are fine from a functional POV.

Anyway, I can certainly keep back this change (or maybe put it on another commit) if it creates any confusion---it was just a spontaneous reaction when my eye fell on this piece of code!

Reply:
Very nice, thanks for clearing it up for me :) Let's go forward with this change, it makes sense.